pandas/core/generic.py

```diff
@@ -3972,6 +3972,16 @@ def asof(self, where, subset=None):
             where = Index(where) if is_list else Index([where])
 
         nulls = self.isnull() if is_series else self[subset].isnull().any(1)
+        if nulls.all():
+            if is_series:
+                return pd.Series(np.nan, index=where)
```

Comment:
I thought about that, @jreback , but when I experimented with a non-null series, I saw that it has no name. I.e.:
```
result = Series(np.random.randn(4), index=[1, 2, 3, 4]).asof([4, 5])
print result
```
returns
```
4   -0.558532
5   -0.558532
dtype: float64
......
```